### DISCUSSION TITLE
"First-class conflicts" should have a brief explanation of the resolution UX, or a link to a page for that

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#2506] [Ideas]
The ["First-class conflicts"](https://github.com/martinvonz/jj/blob/main/docs/conflicts.md) document doesn't really give any clarity about what a user should _do_ once they encounter a conflict. My current understanding is that the easiest and most common way to deal with conflicts is to configure a working merge-editor and then use `jj resolve`, but there may be other ways as well. It would be helpful if the page had a brief explanation of how `resolve` works and/or just a link to the documentation for `resolve`; and if there are other relevant commands than `jj resolve`, those should be mentioned as well.

**Comment by BatmanAoD** [comment]
Possibly it should just link to the ["conflicts" section](https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#conflicts) of the "working copy" document, but I found that a little unclear as well, because it sort of glosses over the actual procedure for considering conflicts resolved:

> To resolve conflicts in a commit, use jj new <commit> to create a working-copy commit on top. You would then have the same conflicts in the working-copy commit. Once you have resolved the conflicts...

I've noticed that in some of the documentation, such as the Git-comparison doc's [command equivalence table](https://github.com/martinvonz/jj/blob/main/docs/git-comparison.md#command-equivalence-table), `echo resolved > filename` is used as an example "resolution". That's supposed to represent manually editing the file to remove conflict markers, right?

**Reply by martinvonz** [reply]
> That's supposed to represent manually editing the file to remove conflict markers, right?

Yes, that's correct.

> My current understanding is that the easiest and most common way to deal with conflicts is to configure a working merge-editor and then use `jj resolve`, but there may be other ways as well.

You can just edit the conflict markers, but I would recommend using a graphical merge editor if possible (I don't because I ssh to my development machine).

**Reply by BatmanAoD** [reply]
Does JJ automatically notice that conflicts are resolved once the markers are removed?

**Reply by martinvonz** [reply]
Exactly. It actually tries to parse them, and if that fails, it assumes the conflict has been resolved. You can actually even resolve only some of the conflicts in a file and continue later.
